StripeFinancialConnections/StripeFinancialConnections/Source/Common/HostViewController.swift

```diff
@@ -106,6 +109,16 @@ extension HostViewController {
             )
             .observe { [weak self] result in
                 guard let self = self else { return }
+
+                analyticsClient.log(
+                    analytic: FinancialConnectionsSheetSynchronizeCompleted(
+                        clientSecret: clientSecret,
+                        class: String(describing: self),
+                        success: result.success
+                    ),
+                    apiClient: .shared
+                )
+
```

Comment:
Apologies in advance for the task not being clear, and it being confusing. I think we can put all the necessary logging in this function

https://github.com/stripe/stripe-ios/blob/9c3d6eb1/StripeFinancialConnections/StripeFinancialConnections/Source/Common/HostViewController.swift#L99-L126

To recap the goal: we lack logging early in the flow so it would be good to know if:
1) first synchronize was called
2) first synchronize succeeded
3) first synchronize error'd

"first synchronize" is not even a great name because it doesn't really matter whether we are calling `synchronize` or some other API call, we just want to know what's going on early in the flow, and we have no way of knowing without more analytics

The current PR theoretically works, but I'm worried about all the extra code that is likely of no big value anyway. For example, we had to add `@_spi(STP) public let analyticsClient: STPAnalyticsClientProtocol` to `STPAPIClient` which is a shared class across SDK's and if it has existed this long without extra dependencies, its likely not a big need to add more.

Reply:
There's at least two ways we could go about modifying this PR.

# Option One (the quick one based off this PR)

1) Keep this `FinancialConnectionsSheetSynchronizeCompleted` here and maybe rename it to something like `FinancialConnectionsSheetInitialSynchronizeCompleted` (but maybe there's better naming like FirstSynchronize or else)
2) Add an extra `FinancialConnectionsSheetInitialSynchronizeStarted` in this same function right before API call is made
3) Delete the rest of the code

That should cover it?

# Option Two (needs research on how to include error details)

Same as Option One except we also figure out how to include error details

This gets tricky because there's some special rules or naming around logging errors so they get picked up by the analytics pipeline/table. [here is an example query that queries the table](https://hubble.corp.stripe.com/queries/kgaidis/63ee3d2d) where the logs go

Reply:
Maybe this is the official way to log errors using AnalyticsClientV1?

https://github.com/stripe/stripe-ios/blob/9c3d6eb1/StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+DeferredAPI.swift#L33-L36

Reply:
Updated to option 1!